### PR TITLE
Javascript/javascript1/week3

### PR DESCRIPTION
**Pull request by JyothiNandyala**
I have added javascript1 week3 homework.

### CODE REVIEW

File: javascript/javascript1/week3/smartphone.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        console.log("You have the limit, you can use smartphone!")
+    }
+
+const totalActivities=activities.length;
+console.log("You have added " +totalActivities+ " activities. They amount to " +usageAmount+ " min. of usage");
+}
+if(activities.length === 0||activities.length===null){
+    console.log("Add some activities before calling showStatus");
+}else{
+    showStatus()
```
Comment (aina21) [inline javascript/javascript1/week3/smartphone.js:38]: why do you call  showStatus() inside showStatus() ?

File: javascript/javascript1/week3/travelInformation.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"use strict"
+
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+  };
+  function returnTime(travelInformation){
+      let time = (travelInformation.destinationDistance)/(travelInformation.speed);
+    var hrs = parseInt(Number(time));
```
Comment (aina21) [inline javascript/javascript1/week3/travelInformation.js:9]: avoid var

**Review by aina21 — CHANGES_REQUESTED** [review]
good job 👍

File: javascript/javascript1/week3/itemArray.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+"use strict"
+
+const names = ["Peter","Ahmad", "Yana","kristina","Rasmus","Samuel","katrine","Tala"];
+  const nameToRemove = names.splice(1, 1);
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/itemArray.js]: yes, iused for loop.

File: javascript/javascript1/week3/savenote.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    const match = notes.find(note => typeof id === 'number' && note.id === id);
+    return match ? match : "ID is not a number and match not found.";
+  }
+  const firstNote = getNote(4);
+console.log(firstNote);   
+
+//find the all content text
+  function logOutNotesFormatted(id) {
+    for(let i=0;i<notes.length;i++){
+        let id =notes[i].id;
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/savenote.js]: yes i changed to const.

File: javascript/javascript1/week3/savenote.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+  logOutNotesFormatted(1); 
+
+  //delete one element from the array of object
+function deleteElementFromObject(id){
+    
+    for(let i=0;i<notes.length;i++){
+        if(notes[i].id===id){
+            const deletedElement=notes.splice(i,1);
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/savenote.js]: actually  don't need that deleted element . i  just created for my purpose but when i was pushing the code i forget to remove,now its removed.

File: javascript/javascript1/week3/seriesduration.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    },
+  ];
+ 
+  const lifespanHours=700800;
+let totalPercentage=0;
+  
+ function logOutSeriesText(){
+    
+    for(let  i=0;i<seriesDurations.length;i++){
+        let totalHours=seriesDurations[i].days*24+seriesDurations[i].hours+(seriesDurations[i].minutes)/60;
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/seriesduration.js]: yes i changed.

File: javascript/javascript1/week3/smartphone.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+"use strict"
+
+//adding activities
+const activities =[];
+var today  = new Date();
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/smartphone.js]: ya,thank you ,i just changed now.

File: javascript/javascript1/week3/smartphone.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        console.log("You have reached your limit, no more smartphoning for you!")
+    }
+    else{
+        console.log("You have the limit, you can use smartphone!")
+    }
+
+const totalActivities=activities.length;
+console.log("You have added " +totalActivities+ " activities. They amount to " +usageAmount+ " min. of usage");
+}
+if(activities.length === 0||activities.length===null){
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/smartphone.js]: here i am checking the null values also , i think its not required, i removed it.

File: javascript/javascript1/week3/travelInformation.js (hunk 2 above)
Comment (JyothiNandyala) [reply javascript/javascript1/week3/travelInformation.js:9]: yes i changed  now.

File: javascript/javascript1/week3/smartphone.js (hunk 1 above)
Comment (JyothiNandyala) [reply javascript/javascript1/week3/smartphone.js]: i changed this.actually i did a mistake.

File: javascript/javascript1/week3/savenote.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("go to walk", 3);
+
+console.log(notes); 
+
+//Retriving the specific id information
+
+function getNote(id) {
+    const match = notes.find(note => typeof id === 'number' && note.id === id);
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/savenote.js:18]: thank you .

File: javascript/javascript1/week3/savenote.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
+"use strict"
+
+const notes = [];
+
+function saveNote(content, id) {
+  notes.push({content, id});
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply javascript/javascript1/week3/savenote.js:6]: hero is an object destructing assignment. That statement defines the variables content and id, then assigns to them the values of properties to notes.content and notes.id correspondingly.

**Comment by JyothiNandyala** [comment]
Hi Ida,

Can i merge the week3 changes.because i did my changes could you please
check and let me know.

**Comment by JyothiNandyala** [comment]
Hi Ida, Can i merge the week3 changes.because i did my changes could you
please check and let me know.
thank you.
… <https://github.com/JyothiNandyala/hyf-homework/pull/5#>

**Review by aina21 — APPROVED** [review]
great :+1:

**Comment by JyothiNandyala** [comment]
week3 homework.

**Pull request merged**